### PR TITLE
Reboot.improve

### PR DESCRIPTION
**Pull request by fmoral2**
#### Proposed Changes ####

-  created a auxiliar method waitInstanceStop to validate the actual stopping and not needing to rely on the output from StopInstances(input) because it can be inconsistent and time wise  and + not needing of eventually function
 
- improve performance with go routines
- added also "reboot on worker nodes"
- added a method for releases Elastic IPs after tests
- added some empty/nil validations
- added validation for the var itself before start terraform 

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

2. Verify code lint; we should not have errors.

3. Update the documentation if needed.

4. Update makefile and docker run if adding new tests.

5. Run your tests at least 4 times with all configurations needed and possible.

6. If needed test with different os types.

### CODE REVIEW

File: pkg/aws/aws.go [hunk 1]
```diff
@@ -146,39 +146,138 @@ func (c Client) DeleteInstance(ip string) error {
...
+		}
+	}
 
-				return nil
+	return "", shared.ReturnLogError("no instances found with the public IP address: ")
+}
+
+func (c Client) StopInstance(instanceID string) error {
+	if instanceID == "" {
+		return shared.ReturnLogError("must sent a instance ID")
```
Comment (ShylajaDevadiga) [inline pkg/aws/aws.go:185]: nit: Can we rephrase this to be more clear? (and other locations)

File: pkg/aws/aws.go [hunk 2]
```diff
@@ -146,39 +146,138 @@ func (c Client) DeleteInstance(ip string) error {
...
+	if startErr != nil {
+		return shared.ReturnLogError("timed out on start instance %s: %v", instanceID, startErr)
+	}
+
+	return nil
+}
+
+func (c Client) ReleaseElasticIps(ipAddress string) error {
+	if ipAddress == "" {
+		return shared.ReturnLogError("must sent an ip address")
```
Comment (ShylajaDevadiga) [inline pkg/aws/aws.go:229]: same

**Review by ShylajaDevadiga — DISMISSED** [review]
small change otherwise LGTM

**Review by mdrahman-suse — APPROVED** [review]
LGTM
